### DISCUSSION TITLE
NewBie Need help in setting up auto-completion and searching functionality

### DISCUSSION THREAD

**Discussion by sarbajit125** [wojciech-kulik/xcodebuild.nvim#234] [Q&A]
Hi,

I just came across this repo and was excited to implement it. With starter config I can run and debug the application. Facing few issue for which need help from  the community. 

1. Our projects use Cocoapods show how to search file/text in Pods directory. The starter Telescope Utils is just searching for files in main directory.
2. How to  print something in DAP console (like po in Xcode console) or clear console before doing some action (like API call)
3. Is there any additional configuration required for LSP, auto-completion seems very basic. Do I need to build it first for better completion ?

Thank you all for creating such a wonderful plugins.

**Comment by wojciech-kulik** [comment]
Hi,

1. You can pass to Telescope commands a path where to search. Please refer to plugin's documentation.
2. You can navigate to dap-repl window and press `a`. Type whatever you would have typed in Xcode.
3. https://github.com/wojciech-kulik/xcodebuild.nvim/wiki/Neovim-Configuration#lsp

**Reply by sarbajit125** [reply]
Hi, @wojciech-kulik  thanks for such a quick response.

1. Yes, I checked plugin documentation and with the help of ChatGpt configured it to search for files in Pods. Issue resolved.
2. Yes just cross checked, debugger is fast than Xcode, but there  is no auto complete in repl window it seems. Still not able to find it's clear command.
3. I have followed the guide and I am getting basic suggestion such as variable and text, but it is unable to find Classes properties mentioned in other file or  calling function with arguments.

**Reply by wojciech-kulik** [reply]
2. I don’t think there is an autocomplete. You can try setting filetype for this buffer to Swift :D maybe it will show something
3. Please show what “:LspInfo” prints and the content of “buildServer.json”

**Reply by sarbajit125** [reply]
2. Ok let me check how to add it.
3. here my LSPInfo

`
==============================================================================
lspconfig: require("lspconfig.health").check()

LSP configs active in this session (globally) ~
- Configured servers: sourcekit
- OK Deprecated servers: (none)

LSP configs active in this buffer (bufnr: 7) ~
- Language client log: ~/.local/state/nvim/lsp.log
- Detected filetype: `swift`
- 2 client(s) attached to this buffer
- Client: `sourcekit` (id: 1, bufnr: [7])
  root directory:    ~/SavingCirclesGIT/SDUI/ios-mobiquity-sdui-core-app/
  filetypes:         swift, objc, objcpp, c, cpp
  cmd:               /usr/bin/sourcekit-lsp
  version:           `?` (Failed to get version) Tried:
  `/usr/bin/sourcekit-lsp --version`
  `/usr/bin/sourcekit-lsp -version`
  `/usr/bin/sourcekit-lsp version`
  `/usr/bin/sourcekit-lsp --help`
  
  executable:        true
  autostart:         true
- Client: `copilot` (id: 2, bufnr: [7])
  root directory:    ~/SavingCirclesGIT/SDUI/ios-mobiquity-sdui-core-app/
  filetypes:         
  cmd:               ~/SavingCirclesGIT/SDUI/ios-mobiquity-sdui-core-app/node /Users/comviva/.local/share/nvim/lazy/copilot.lua/copilot/index.js
  version:           `v23.1.0`
  executable:        true
  autostart:         false

Docs for active configs: ~
- sourcekit docs: >markdown
  
  https://github.com/apple/sourcekit-lsp
  
  Language server for Swift and C/C++/Objective-C.
      
  
- 

`

BuildServer.Json

`{
	"name": "xcode build server",
	"version": "0.2",
	"bspVersion": "2.0",
	"languages": [
		"c",
		"cpp",
		"objective-c",
		"objective-cpp",
		"swift"
	],
	"argv": [
		"/opt/homebrew/bin/xcode-build-server"
	],
	"workspace": "/Users/comviva/SavingCirclesGIT/SDUI/ios-mobiquity-sdui-core-app/ios-mobiquity-sdui-core-app.xcworkspace",
	"build_root": "/Users/comviva/Library/Developer/Xcode/DerivedData/ios-mobiquity-sdui-core-app-bjtdmqspfzdvzjcwagokyuxagwha",
	"scheme": "MOCK",
	"kind": "xcode"
}`

**Reply by wojciech-kulik** [reply]
2. I tried it by setting `:set ft=swift` inside `dap-repl`, it works partially, but it's not very stable I think :)
3. Looks good, I'm not sure what could be the problem.  Did you install Xcode in Applications directory? That could help. Also make sure that command `xcode-select -p` points to the same Xcode app.

Also, I see that you are using `/usr/bin/sourcekit-lsp` maybe try with the one from Xcode. In your LSP setup function you can set 
```lua
cmd = { vim.trim(vim.fn.system("xcrun -f sourcekit-lsp")) }
```

You can also run `:LspLog` to see if there is anything meaningful there.

**Reply by sarbajit125** [reply]
@wojciech-kulik , The LSP issues seem inconsistent, sometimes it is working fine, and some it is not able to load suggestions. Thanks for your help.  Need your suggestion on two topics.

1. I am working on large UIKit project. and use heirchy feature of code to know which View controller is this and navigate there. How you suggest to do this in Nvim.

2. Also when I am searching anything in telescope when DAP is running entire layout is getting distorted. Searched page instead of occupying maximum size is going to a small pane or disappearing itself.

**Reply by wojciech-kulik** [reply]
1. I'm not sure what do you mean by "hierarchy feature". I guess a code tree for a file? There are several plugins to do that. I don't remember names, you can google that.
2. Use this config and see if it happens with it: https://github.com/wojciech-kulik/ios-dev-starter-nvim If not then try debugging what differs in your config.

**Reply by sarbajit125** [reply]
1. By hierarchy Feature I mean debug hierarchy option which show the layout and viewController name. Attaching Image below for reference.
[Debug Hierarchy](https://i.sstatic.net/emVdB.jpg)
2. Yes I am following starter template only. Attaching screenshot  below for better understanding.
1st one is default layout.
<img width="1440" alt="Screenshot 2024-11-17 at 8 56 24 PM" src="https://github.com/user-attachments/assets/952fce8c-47f4-4051-8496-7ac8a37e2545">
2nd one is after using telescope to find file while DAP is running. As you can code file has move to left bottom instead of occupying center space
<img width="1440" alt="Screenshot 2024-11-17 at 8 56 39 PM" src="https://github.com/user-attachments/assets/d52709f7-0745-4566-9bfc-9f282bf5aca6">

**Reply by wojciech-kulik** [reply]
1. If you don't want to use Xcode, there are some standalone alternatives: https://github.com/wojciech-kulik/xcodebuild.nvim/issues/213
2. I'm not sure why it happens, the plugin most likely doesn't change anything that could cause that.

**Reply by sarbajit125** [reply]
1. Yes I am aware of that, but it is a subscription-based service. How do you find files when debugging in a large project.
2. Is there any way to lock the layout?. I added vim.api.nvim_set_keymap('n', '<leader>du', '<cmd>lua require("dapui").toggle({ reset = true })<CR>', { noremap = true, silent = true }) to keymap still it does not reset many time.

**Reply by wojciech-kulik** [reply]
1. By naming them in a meaningful way & structuring the folder hierarchy correctly. In general good coding practices :). When working on a project that I don't know I also use NvimTree for easier navigation. When I know the project better then I use Telescope most of the time.
